TypeSystem/flow-analysis/promotion_via_type_test_A03_t07.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a variable `x` is promotable via type test with type
+/// `T` given variable model `VM` if
+/// - `VM = VariableModel(declared, promoted, tested, assigned, unassigned,
+///     captured)`
+/// - and `captured` is `false`
+/// - and `S` is the current type of `x` in `VM`
+/// - and not `S <: T`
+/// - and `T <: S` or (`S` is `X extends R` and `T <: R`) or (`S` is `X & R` and
+///   `T <: R`)
+///
+/// @description Checks that a variable is promoted if `T` is a subtype of `S`.
+/// Test type `Object?` as `T`.
+/// @author sgrekhov22@gmail.com
+
+import '../../Utils/static_type_helper.dart';
+
+main() {
+  dynamic d = "a";
+  if (d is Object?) { // ignore: unnecessary_type_check
+    d.expectStaticType<Exactly<Object?>>();
```

Comment:
We can't promote from `dynamic` to `Object?`, but this test won't tell us that `d` still has the type `dynamic` _before run time_ (it will throw because there is no such method).

Perhaps use `if (1 > 2) d.testDynamic;` instead?